workflow-parser/src/model/converter/workflow-call.ts

```diff
@@ -27,6 +29,95 @@ export function convertEventWorkflowCall(context: TemplateContext, token: Mappin
   return result;
 }
 
+
+export function convertWorkflowInputs(
+  context: TemplateContext,
+  token: MappingToken
+): {
+  [inputName: string]: InputConfig;
+} {
+  const result: {[inputName: string]: InputConfig} = {};
+
+  for (const item of token) {
+    const inputName = item.key.assertString("input name");
+    const inputMapping = item.value.assertMapping("input configuration");
+
+    result[inputName.value] = convertWorkflowInput(context, inputMapping);
+  }
+
+  return result;
+}
+
+export function convertWorkflowInput(context: TemplateContext, token: MappingToken): InputConfig {
+  const result: InputConfig = {
+    type: InputType.string // Default to string
+  };
+
+  let defaultValue: undefined | ScalarToken;
+
+  for (const item of token) {
+    const key = item.key.assertString("workflow dispatch input key");
+
+    switch (key.value) {
+      case "description":
+        result.description = item.value.assertString("input description").value;
+        break;
+
+      case "required":
+        result.required = item.value.assertBoolean("input required").value;
+        break;
+
+      case "default":
+        defaultValue = item.value.assertScalar("input default");
+        break;
+
+      case "type":
+        result.type = item.value.assertScalar("input type");
+        break;
```

Comment:
@felipesu19 this is why you were having issues with the type I suggested:

```
Type 'ScalarToken' is not assignable to type 'InputType'.
```

I don't think `type` here needs to support scalars:

```json
    "workflow-call-input-type": {
      "description": "Required if input is defined for the `on.workflow_call` keyword. The value of this parameter is a string specifying the data type of the input. This must be one of: `boolean`, `number`, or `string`.",
      "one-of": [
        "input-type-string",
        "input-type-boolean",
        "input-type-number"
      ]
    },
    "input-type-string": {
      "string": {
        "constant": "string"
      }
    },
    "input-type-boolean": {
      "string": {
        "constant": "boolean"
      }
    },
    "input-type-number": {
      "string": {
        "constant": "number"
      }
    },
```